cocotb/handle.py

```diff
@@ -530,11 +519,19 @@ def _range_iter(self, left, right):
 
     @NonHierarchyObject.value.getter
     def value(self):
-        # need to iterate over the sub-object
-        result = []
-        for x in range(len(self)):
-            result.append(self[x].value)
-        return result
+        # need to iterate over the sub-objects
+        return [sub_handle.value for sub_handle in self]
+
+    @value.setter
+    def value(self, value):
+        if type(value) is list:
+            if len(value) != len(self):
+                raise IndexError("Assigning list of length %d to object %s of length %d" % (
+                    len(value), self._name, len(self)))
+            for val_idx, self_idx in enumerate(self._range_iter(self._range[0], self._range[1])):
+                self[self_idx] = value[val_idx]
+        else:
+            raise TypeError("Assigning non-list value to object %s of type %s" % (self._name, type(self)))
```

Comment:
Indentation is shorter as:

```suggestion
        if type(value) is not list:
            raise TypeError("Assigning non-list value to object %s of type %s" % (self._name, type(self)))
        if len(value) != len(self):
            raise IndexError("Assigning list of length %d to object %s of length %d" % (
                len(value), self._name, len(self)))
        for val_idx, self_idx in enumerate(self._range_iter(self._range[0], self._range[1])):
            self[self_idx] = value[val_idx]
```